python/lsst/ap/verify/dataset.py

```diff
@@ -90,20 +90,27 @@ def __init__(self, datasetId):
         except KeyError:
             raise ValueError('Unsupported dataset: ' + datasetId)
 
-        self._dataRootDir = getPackageDir(datasetPackage)
-        self._validatePackage()
+        try:
+            self._dataRootDir = getPackageDir(datasetPackage)
+            self._validatePackage()
+        except pexExcept.NotFoundError as e:
```

Comment:
Can `_validatePackage` throw a `pexExcept.NotFoundError`? If it can (which I'm dubious about), would it mean that the package hasn't been set up, as the `RuntimeError` below claims?

Reply:
It's getPackageDir that is throwing the NotFoundError if the package is not set up.

Reply:
Yes, of course. Hence my comment indicate that there's no point having the `except` block around `_validatePackage`...

Reply:
I put them in the same `try` block because `self._dataRootDir` being initialized is a precondition for `_validatePackage`, and this way it's easier to see that the precondition will always be satisfied. As noted, `_validatePackage()` will never raise `NotFoundError`, so the exception handling is unambiguous.

Reply:
The usual recommendation is to use an `else` block with the `try`. Fluent Python makes an explicit case for doing that (the `else` only triggers if the try block completed).

Reply:
Fine, I'll do that.